### PR TITLE
feat comment mention

### PR DESCRIPTION
**Pull request by Yxwww**
- Add MentionView
- Improve Mention styles
- Add useHover and mention calculation
- Update MentionView with simplied logic
- Replace suggestion with autocomplete plugin
- Refactor MentionLilst
- [x] cancel on first space not working properly for suggestions ? 

Before merge:
- [x] ensure yarn is deduped

### CODE REVIEW

File: src/prosemirror/plugins/index.ts [hunk 1]
```diff
@@ -44,12 +46,12 @@ export function getPlugins(
...
-        (action) => store.dispatch(handleSuggestion(action)),
-        NO_VARIABLE,
-        // Cancel on space after some of the triggers
-        (trigger) => !trigger?.match(/(?:(?:[a-zA-Z0-9_]+)\s?=)|(?:\{\{)/),
-      ),
+      // ...suggestion(
+      //   (action) => store.dispatch(handleSuggestion(action)),
+      //   NO_VARIABLE,
+      //   // Cancel on space after some of the triggers
+      //   (trigger) => !trigger?.match(/(?:(?:[a-zA-Z0-9_]+)\s?=)|(?:\{\{)/),
```
Comment (Yxwww) [inline src/prosemirror/plugins/index.ts:53]: TODO: revive this logic?

File: src/prosemirror/plugins/index.ts [hunk 2]
```diff
@@ -60,14 +61,20 @@ export function getPlugins(
...
-      (trigger) => !trigger?.match(/(?:(?:[a-zA-Z0-9_]+)\s?=)|(?:\{\{)/),
-    ),
-    commentsPlugin(),
     getPromptPlugin(),
+    ...autocomplete({
+      triggers: [
+        // { name: 'mention', trigger: '@', cancelOnFirstSpace: false },
+        {
+          name: 'suggestion',
+          trigger: schema.nodes.variable ? ALL_TRIGGERS : NO_VARIABLE,
```
Comment (rowanc1) [inline src/prosemirror/plugins/index.ts:70]: We can probably break this out into a bunch of different triggers. Some of them canceled on space and some did not.

File: src/prosemirror/plugins/suggestion.ts [hunk 3]
```diff
@@ -1,9 +1,10 @@
-import { Plugin, PluginKey, Selection } from 'prosemirror-state';
+import { Plugin, PluginKey } from 'prosemirror-state';
```
Comment (rowanc1) [inline src/prosemirror/plugins/suggestion.ts:1]: Can we just completely delete this file?

File: src/store/suggestion/types.ts [hunk 4]
```diff
@@ -6,6 +6,7 @@ import { CommandResult } from './commands';
 export const UPDATE_SUGGESTION = 'UPDATE_SUGGESTION';
 export const UPDATE_RESULTS = 'UPDATE_RESULTS';
 export const SELECT_SUGGESTION = 'SELECT_SUGGESTION';
+export const UPDATE_SUGGESTION_DATA = 'UPDATE_SUGGESTION_DATA';
```
Comment (rowanc1) [inline src/store/suggestion/types.ts:9]: Can we delete some of this stuff?

File: yarn.lock [hunk 5]
```diff
@@ -4654,6 +4663,15 @@ prosemirror-view@^1.0.0, prosemirror-view@^1.1.0, prosemirror-view@^1.13.3, pros
     prosemirror-state "^1.0.0"
     prosemirror-transform "^1.1.0"
 
+prosemirror-view@^1.20.3:
```
Comment (rowanc1) [inline yarn.lock:4666]: Can we dedup this yarn lock please!

**Review by rowanc1 — COMMENTED** [review]
A few notes on the autocomplete lib.

**Comment by rowanc1** [comment]
@Yxwww I have opened up #109 that can come into here after your review.

It requires a few of the updates in https://github.com/curvenote/schema/pull/36

File: src/prosemirror/plugins/suggestion.ts (hunk 3 above)
Comment (rowanc1) [reply src/prosemirror/plugins/suggestion.ts:1]: I have deleted this in the branch.

File: yarn.lock (hunk 5 above)
Comment (rowanc1) [reply yarn.lock:4666]: I did this in #109

File: src/components/Editor.tsx [hunk 6]
```diff
@@ -5,6 +5,7 @@ import { EditorState, Transaction } from 'prosemirror-state';
 import { opts } from '../connect';
 import { createEditorView } from '../prosemirror';
 import { Dispatch, State, actions, selectors } from '../store';
+import { Schema } from 'prosemirror-model';
```
Comment (rowanc1) [inline src/components/Editor.tsx:8]: This isn't used.

File: src/prosemirror/index.ts [hunk 7]
```diff
@@ -75,6 +75,9 @@ export function createEditorView(
         dynamic: views.newWidgetView,
         range: views.newWidgetView,
         switch: views.newWidgetView,
+        mention(node, view, getPos) {
+          return new views.MentionView(node, view, getPos as GetPos);
+        },
```
Comment (rowanc1) [inline src/prosemirror/index.ts:80]: Can we follow the same style of `mention: views.MentionView`?

File: src/prosemirror/plugins/index.ts [hunk 8]
```diff
@@ -44,14 +42,14 @@ export function getPlugins(
...
-        (trigger) => !trigger?.match(/(?:(?:[a-zA-Z0-9_]+)\s?=)|(?:\{\{)/),
-      ),
-      ...inputrules(schema),
       keymap(buildCommentKeymap(stateKey, schema)),
+      ...autocomplete({
+        triggers: [{ name: 'mention', trigger: '@', cancelOnFirstSpace: false }],
+        reducer(action) {
+          return store.dispatch(handleSuggestion(action));
+        },
+      }),
```
Comment (rowanc1) [inline src/prosemirror/plugins/index.ts:51]: I think we need to have the other triggers back in here.

**Pull request merged**